community/cypher/frontend-3.2/src/main/scala/org/neo4j/cypher/internal/frontend/v3_2/symbols/IntegerType.scala

```diff
@@ -22,7 +22,7 @@ package org.neo4j.cypher.internal.frontend.v3_2.symbols
 object IntegerType {
   val instance = new IntegerType() {
     val parentType = CTNumber
-    override lazy val coercibleTo: Set[CypherType] = Set(CTFloat)
+    override lazy val coercibleTo: Set[CypherType] = Set(CTFloat) ++ parentType.coercibleTo
```

Comment:
Neither `NumberType` nor `AnyType` do override `coercibleTo`, so the ` ++ parentType.coercibleTo` will add an empty list from the `CypherType` implementation, i.e. have no effect. What was the reason for adding this? Is it if anyone later changes any of the superclasses' implementations?

Reply:
Excatly. The idea behind `coercibleTo` is to specify your coercions and then add those of all parent types (which are empty as of now). I actually forgot that in `ListType`, gonna add it there too.